HW2/P2/parallel_vector.pyx

```diff
@@ -80,12 +80,45 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    # data movement.  Be sure to avoid deadlock, and double-locking.
    ##########
    with nogil:
-       for r in range(repeat):
+       for r in prange(repeat, num_threads=4):
```

Comment:
I wonder if there's a difference to the compiler between declaring nogil outside or with the prange.